app/Jobs/RequeuePendingQsBatchesJob.php

```diff
@@ -17,18 +18,20 @@ public function __construct() {
 
     public function handle(): void
     {
-        tap(QsBatch::where([
-            ['processing_attempts', '>=', $this->markFailedAfter],
-            ['failed', '=', false],
-        ]))
-            ->update(['failed' => true])
-            ->get()
-            ->each(function ($batch, $index) {
-                report("QsBatch with ID ".$batch->id."was marked as failed.");
-            });
+        DB::transaction(function () {
```

Comment:
Wrapped this in a transaction so we get locking and don't run into funny races somehow.